### PR TITLE
Eos 26606: Improve logging for hare mini-provisioning

### PR DESCRIPTION
**Pull request by ydb242**
Problem: Current logs are not enough to understand the root cause in many cases.
Solution:
    a) Update the logging.debug statements to logging.info to increase visibility
    b) Add extra log statements for important functions

### CODE REVIEW

File: provisioning/miniprov/hare_mp/main.py [hunk 1]
```diff
@@ -452,22 +495,27 @@ def start_mkfs_parallel(hostname: str, hare_config_dir: str):
...
 
 
 @repeat_if_fails()
 def is_mkfs_done_on_all_nodes(utils: Utils,
                               cns_utils: ConsulUtil,
                               nodes: List[str]) -> bool:
+    logging.info("Entering " + is_mkfs_done_on_all_nodes.__name__)
     for node in nodes:
         if not cns_utils.kv.kv_get(f'mkfs_done/{node}', recurse=True):
             return False
```
Comment (SwapnilGaonkar7) [inline provisioning/miniprov/hare_mp/main.py:510]: Can we print 'node' here before returning False

File: provisioning/miniprov/hare_mp/main.py [hunk 2]
```diff
@@ -513,6 +566,7 @@ def init(args):
         # Stopping hax and consul
         hax_starter.stop()
```
Comment (SwapnilGaonkar7) [inline provisioning/miniprov/hare_mp/main.py:567]: Check if hax and consul stop are getting logged, you can add logs in there respective functions which are present in 'consul_starter.py' and 'hax_starter.py'

File: provisioning/miniprov/hare_mp/main.py [hunk 3]
```diff
@@ -476,13 +524,18 @@ def cleanup_mkfs_state(utils: Utils, cns_utils: ConsulUtil):
...
 def set_mkfs_done_for(node: str, cns_utils: ConsulUtil):
+    logging.info("Entering " + set_mkfs_done_for.__name__)
     cns_utils.kv.kv_put(f'mkfs_done/{node}', 'true')
+    logging.info("Exiting " + set_mkfs_done_for.__name__)
 
 
 def init(args):
+    logging.info("Entering " + init.__name__)
     try:
         url = args.config[0]
```
Comment (SwapnilGaonkar7) [inline provisioning/miniprov/hare_mp/main.py:541]: Let's log incase `if not is_mkfs_required(url):`
We are exiting here so better to highlight here.

**Comment by mssawant** [comment]
@ydb242, instead of having logging calls to log just the function names, we can have a decorators as,
```
def func_enter():
    """
    Logs function entry point.
    """
    def callable(f):
        @wraps(f)
        def wrapper(*args, **kwds):
            func_name = f.__qualname__
            Log.info('Entering %s', func_name)
            return f(*args, **kwds)
        return wrapper

    return callable

def func_leave():
    """
    Logs function entry point.
    """
    def callable(f):
        @wraps(f)
        def wrapper(*args, **kwds):
            func_name = f.__qualname__
            result = f(*args, **kwds)
            Log.info('Leaving %s', func_name)
            return result
        return wrapper

    return callable
```
call them as,

```
@func_enter
def foo()

@func_leave
def bar()

@func_enter()
@func_leave()
def baz()
```

**Comment by suvratjoshi** [comment]
Hi @ydb242 
Please modify commit message to limit number of chars to 74 per line.
Thanks!

**Comment by suvratjoshi** [comment]
I think we need to have loggig/entry/exit in 'provisioning\miniprov\hare_mp\cdf.py' file too.

**Comment by suvratjoshi** [comment]
@ydb242 
Can you please attach the logs file / post contents of logs file here.
Also please confirm if both deployment methods are passed.

Thanks!

**Comment by ydb242** [comment]
@suvratjoshi 
> I think we need to have logging/entry/exit in 'provisioning\miniprov\hare_mp\cdf.py' file too.

Yes, I will make these changes.
> Hi @ydb242 Please modify commit message to limit number of chars to 74 per line. Thanks!

Ok, I will update the commit message.

> @ydb242 Can you please attach the logs file / post contents of logs file here. Also please confirm if both deployment methods are passed.
> 
> Thanks!

Currently for some reason, the logs are not getting populated. We are seeing that the setup.log file is empty.
Working on resolving the issue.
Once it is resolved, I will update the PR with the necessary outputs and move it from draft to actual PR.

**Comment by suvratjoshi** [comment]
Hi @ydb242 
Also, can you try this out if this kind of formatting can be applied to logs here in mini provisioner, if easily possible
@Shreya-18 had tried this once and had worked for her, in cfgen. @Shreya-18 can confirm and help, if required.
```
[10/8/2021 5:30 PM] Shreya Karmakar
LOG = logging.getLogger('cfgen')

[10/8/2021 5:30 PM] Shreya Karmakar
%(pathname)s Full pathname of the source file where the logging call was issued(if available).
%(filename)s Filename portion of pathname.%(module)s Module (name portion of filename).
%(funcName)s Name of function containing the logging call.
%(lineno)d Source line number where the logging call was issued (if available).

[10/8/2021 5:31 PM] Shreya Karmakar
formatter = logging.Formatter('[%(asctime)s] p%(process)s {%(pathname)s:%(lineno)d} %(levelname)s - %(message)s','%m-%d %H:%M:%S')
```

Also please try removing codacy errors using the suggestion, populated by codacy.
Thanks!

**Comment by Shreya-18** [comment]
> Hi @ydb242 Also, can you try this out if this kind of formatting can be applied to logs here in mini provisioner, if easily possible @Shreya-18 had tried this once and had worked for her, in cfgen. @Shreya-18 can confirm and help, if required.
> 
> ```
> [10/8/2021 5:30 PM] Shreya Karmakar
> LOG = logging.getLogger('cfgen')
> 
> [10/8/2021 5:30 PM] Shreya Karmakar
> %(pathname)s Full pathname of the source file where the logging call was issued(if available).
> %(filename)s Filename portion of pathname.%(module)s Module (name portion of filename).
> %(funcName)s Name of function containing the logging call.
> %(lineno)d Source line number where the logging call was issued (if available).
> 
> [10/8/2021 5:31 PM] Shreya Karmakar
> formatter = logging.Formatter('[%(asctime)s] p%(process)s {%(pathname)s:%(lineno)d} %(levelname)s - %(message)s','%m-%d %H:%M:%S')
> ```
Yes.. @ydb242  similar to motr ERROR logs, s3 INFO logs
```
motr[00014]:  ec80  ERROR  [helpers/ufid.c:263:m0_ufid_init]  -75: Can not generate service id

 s3server.cc:196] [dispatch_s3_api_request] [ReqID: -] Req uri [/]

```
here we see **`<file_path>:<source line number>:<function name>`** ...which helps a lot in debugging (by adding a 1 liner change)

**Comment by ydb242** [comment]
Closing this PR and raising another one soon

**Pull request closed without merging**